### PR TITLE
Rs 663

### PR DESCRIPTION
**Pull request by aChanEP**
Description:

Builds a general component `PaymentSelectorMain` that renders payment options for all versions of EP in both the profile and orders resource.  Check out the component README for more details: `src/components/PaymentSelectorMain/README.md`.

Deletes `ProfilePaymentMethods` in place of the aforementioned component.

The `CheckoutPage.tsx` and `ProfilePage.tsx` both integrate this component.

NOTE:  Ability to select this component will be added in future PR's to accomodate for PO number entry designs.

Linting:

- [x] No linting errors

Component Updates:

- [x] Component package requires update on npm [@elasticpath/store-components](https://www.npmjs.com/package/@elasticpath/store-components)

Tests:

- [x] E2E tests (npm test run with `e2e`)
- [x] Manual tests

1. connect to `https://reference.elasticpath.com` and go through checkout flow and payment options in profile
2. connect to the latest payments release and do the same.  Notice the application functions for both.
3. Checkout storybook stories.  
4. Read through storybook documentation.

Documentation:

- [ ] Requires documentation updates

### CODE REVIEW

**Comment by aChanEP** [comment]
Just found a bug when deleting the currently selected default payment method!  Updating!

File: src/components/src/PaymentSelectorMain/README.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
+# PaymentSelectorMain
+
+#### Description
+
+Displays payment selector for all versions of Cortex payments in both the Orders and Profiles resource.  Look for example usage in the CheckoutPage.tsx and ProfilePage.tsx or in storybook stories.
+
+#### How to use component with Cortex 7.6 Payments update
```
Comment (shaunmaharaj) [inline src/components/src/PaymentSelectorMain/README.md]: wonderful!!!

File: src/components/src/PaymentSelectorMain/README.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+defaultprofile:paymentinstruments:defaultinstrumentselector,
+defaultprofile:paymentinstruments:defaultinstrumentselector:chosen,
+defaultprofile:paymentinstruments:defaultinstrumentselector:chosen:description,
+defaultprofile:paymentinstruments:defaultinstrumentselector:choice,
+defaultprofile:paymentinstruments:defaultinstrumentselector:choice:description,
+```
+
+`shouldPostToProfile` as well as `showSaveToProfileOption` are props surfaced up from the `paymentform.main.tsx` component.  Refer to that `PaymentForm/README.md` for more information.
+
+`onChange` is a calledback that is called when either the a delete button, a radio button or `paymentform.main.tsx` component is submitted.  It is designed for the parent to make another request to Cortex and update its state and the state of its child, `PaymentSelectorMain`.
```
Comment (shaunmaharaj) [inline src/components/src/PaymentSelectorMain/README.md]:
```suggestion
`onChange` is a calledback that is called when either a delete button, a radio button or `paymentform.main.tsx` component is submitted.  It is designed for the parent to make another request to Cortex and update its state and the state of its child, `PaymentSelectorMain`.
```

File: src/components/src/PaymentSelectorMain/paymentselector.main.tsx [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+      allPaymentInstruments = [...paymentInstrumentSelector._choice, { ...paymentInstrumentSelector._chosen[0], chosen: true }];
+    } else if (paymentInstrumentSelector._chosen) {
+      allPaymentInstruments = [{ ...paymentInstrumentSelector._chosen[0], chosen: true }];
+    } else if (paymentInstrumentSelector._choice) {
+      allPaymentInstruments = [...paymentInstrumentSelector._choice];
+    } else {
+      return null;
+    }
+
+    return allPaymentInstruments.sort((paymentInstrumentA, paymentInstrumentB) => {
```
Comment (shaunmaharaj) [inline src/components/src/PaymentSelectorMain/paymentselector.main.tsx]: Is this sorting necessary?

File: src/components/src/PaymentSelectorMain/paymentselector.main.tsx [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+
+    return (
+      paymentMethods.map(payment => this.renderPaymentChoice(payment))
+    );
+  }
+
+  renderPayments() {
+    const { paymentMethods, paymentInstrumentSelector, paymentMethodInfo } = this.props;
+
+    if (paymentInstrumentSelector) {
```
Comment (shaunmaharaj) [inline src/components/src/PaymentSelectorMain/paymentselector.main.tsx]: Clean usage for separation of functions 👍

**Pull request merged**